### DISCUSSION TITLE
Canteen (Refills in general) crafting state change.

### DISCUSSION THREAD

**Discussion by Ongjming** [SmartlyDressedGames/Unturned-3.x-Community#2516] [Ideas]
Was wondering is there currently anyway for someone to craft a recipe using a specific condition of the canteen and at the end of the crafting returns another specific condition for the canteen?
Was thinking of this because when you craft portable fuel can into industrial ones, they retain their fuel. (based on this line I saw on Industrial fuel can "Blueprint_0_State_Transfer")

E.g.
Raw Venison + Canteen (With salty water) = Salted venison + Canteen (Empty)
Canteen (Clean water) + Salt = Canteen (Salty water)

I thought about using it as a tool in recipes but its state wouldn't change and won't really be checked for in the recipe.
Using it in craft recipe as it is, also produce the same issue as above, it would end up returning you an empty Canteen despite its condition prior.
Purification tablet kinda changes the state of Dirty and Salty water to Clean, but in the process you cannot really create anything else.

Other than bottled water, there isn't really any pure water source out there.

**Comment by SDGNelson** [comment]
Interesting idea, but you are right sorry there is no setting for this yet.
